### DISCUSSION TITLE
Does anyone knows why in iOS, the webview, although is instructed to call an url in which all the "+" signs were URL escaped , changes them back to "+" before sending the requests to the backend server?

### DISCUSSION THREAD

**Discussion by ghevge** [apache/cordova#492] [Q&A] [answered]
Does anyone knows why in iOS, the webview, although is instructed to call an url in which all the "+" signs were URL escaped , changes them back to "+" before sending the requests to the backend server?

I'm calling the backed server as follows:

```
settings["mode"] = 'cors';
settings["credentials"] = 'same-origin';

    try {
        const response = await fetch(url, settings);
```
the url passed to the fetch command is "https://www.mydomain.com/search?string=%2Bdistance"
But when I inspect the safari network tab, webview is converting this url into: "https://www.mydomain.com/search?string=+distance"
I am seeing the same garbage at my server side proxy level, so is not just an illusion visible in safari network tab. 

Unfortunately by passing the  unescaped "+" causes problem at by backed java server level, which interprets the "+" as an empty space, messing up my logic.

This weird behavior is only visible in iOS. 
Android and web browsers are working just fine with the exact same code.

Any idea what is causing this iOS behavior, and is there a way to fix it ?

Thanks

PS: I am using the latest cordova version
ios@7.1.1

**Comment by BigBalli** [comment]
Didn't have time to look into why... but a temporary fix would be to:
- double encode the plus signs: Instead of just encoding "+" to "%2B", encode it to "%252B". This way, when the WebView decodes it once, it will still be "%2B".
- use a plugin that leverages native for external calls

FYI I just tried both using  "+" and "%2B" and they're left intact when inspected in Web Inspector.

**Comment by ghevge** [comment]
Thanks for your feedback! 

I just realized that I forgot about the cordova-plugin-webview-proxy I am using only for ios.

I think the problem comes from somewhere in that plugin... 

I've also tried the double encoding, but funny enough when I do that, it just passes to the up stream the double encoded string, it doesn't decode it once.

I will try to see if I can trick it to pass what it has to pass to the upstream.

**Comment by ghevge** [comment, marked as answer]
I can confirm that the problem is coming from cordova-plugin-webview-proxy. 
Sorry for the hassle!
